### DISCUSSION TITLE
the "i just wanted to be sure" batch of questions

### DISCUSSION THREAD

**Discussion by mn-pal-m** [slint-ui/slint#4289] [Q&A] [answered]
how am i able to, or is it possible to do any of the following:
- 1 - altering the string-literal values of `@image-url` and/or being able to assign an image included in-build (so no relative/absolute path, so that the user does not requires having the individual image assets when downloading the executable) for dynamic insertion based upon a callback behavior status? 
- 2 - specify a callback to be executed on program execution (so as soon as the slint backend successfully opens)
- 3 - although certainly not something i have in plans at the moment, the ability to specify a region that can be used as a window drag (so for example as a title bar alternative)

i really just wanted to be sure, as even checking in some of the threads i was still left one way or another with a big question mark in those regards. i appreciate it! ^^

**Comment by ogoffart** [comment, marked as answer]
> altering the string-literal values of @image-url and/or being able to assign an image included in-build 

You can get an image loaded by native code.

For exemple:

```
export global Logic {
   callback load-image(string) -> image;
}

//...

    Image { 
        source: Logic.load-image(some-url-property);
    }
```

Then you implement the load-image in native code using the slint::Image API.

> specify a callback to be executed on program execution (so as soon as the slint backend successfully opens)

I believe you can use the `init` callback https://slint.dev/releases/1.3.2/docs/slint/src/language/builtins/callbacks#init

> although certainly not something i have in plans at the moment, the ability to specify a region that can be used as a window drag (so for example as a title bar alternative)

Not yet implemented. But tracked in https://github.com/slint-ui/slint/issues/613
